File: app/src/main/java/com/example/nijimac103/itunestracker/service/view/VideoViewActivity.java
package com.example.nijimac103.itunestracker.service.view;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;

import com.example.nijimac103.itunestracker.R;
import com.example.nijimac103.itunestracker.service.util.FragmentUtils;
import com.example.nijimac103.itunestracker.service.view.Fragment.ArtistVideoViewFragment;

import static com.example.nijimac103.itunestracker.service.view.MainActivity.URL;

public class VideoViewActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_video_view);

        ArtistVideoViewFragment fragment = new ArtistVideoViewFragment();

        Intent i = getIntent();
        String artistPreviewUrl = i.getStringExtra(URL);

        addVideoViewFragment(fragment, artistPreviewUrl);

    }

    private void addVideoViewFragment(ArtistVideoViewFragment fragment, String artistPreviewUrl) {

        FragmentUtils.setArgsToFragment(fragment, URL,artistPreviewUrl);

        FragmentUtils.insertFragmentToActivity(R.id.fragment_container,getSupportFragmentManager(),fragment,ArtistVideoViewFragment.TAG);
    }

}

File: app/src/main/java/jp/co/geisha/itunestracker/view/activity/VideoViewActivity.kt
package jp.co.geisha.itunestracker.view.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

import jp.co.geisha.itunestracker.R
import jp.co.geisha.itunestracker.util.FragmentUtils
import jp.co.geisha.itunestracker.view.fragment.ArtistVideoViewFragment

import jp.co.geisha.itunestracker.ARTIST_VIDEO_VIEW_FRAGMENT_TAG
import jp.co.geisha.itunestracker.URL

class VideoViewActivity : AppCompatActivity() {

    companion object {
        fun createIntent(context: Context, url: String): Intent {
            return Intent(context, VideoViewActivity::class.java).apply {
                putExtra(URL, url)
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_video_view)

        val artistPreviewUrl = intent?.getStringExtra(URL)

        artistPreviewUrl?.let {
            addVideoViewFragment(ArtistVideoViewFragment(), artistPreviewUrl)
        } ?: run {
            finish()
        }
    }

    private fun addVideoViewFragment(fragment: ArtistVideoViewFragment, artistPreviewUrl: String) {
        FragmentUtils.run {
            setArgsToFragment(fragment, URL, artistPreviewUrl)
            insertFragmentToActivity(R.id.fragmentContainer, supportFragmentManager, fragment, ARTIST_VIDEO_VIEW_FRAGMENT_TAG)
        }
    }
}
